pkg/controller/utils/component.go

```diff
@@ -142,6 +247,9 @@ func (c *componentHandler) createOrUpdateObject(ctx context.Context, obj client.
 	// Check to see if the object exists or not - this determines whether we should create or update.
 	err := c.client.Get(ctx, key, cur)
 	if err != nil {
+		// Invalidate our cached object.
+		dCache.delete(obj)
```

Comment:
Wondering why we have this cache invalidation here, rather than after the `errors.IsNotFound` check?

Reply:
My main thought was for scenarios where there is some kind of transient error that resolves on retry. In that case, we would be creating a potential cache miss that seems unnecessary. Happy to leave this choice up to you